bob-backend/src/pearl/data.rs

```diff
@@ -2,19 +2,27 @@ use crate::prelude::*;
 
 include!(concat!(env!("OUT_DIR"), "/key_constants.rs"));
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
Where is the implementation of the comparison function itself?

Reply:
It is derived with this traits. You mean that we should compare bytes from right to left?

Reply:
Yes. The comparison should match Little Endian order (from right to left)

Reply:
Fixed cmp order